nvflare/app_common/executors/learner_executor.py

```diff
@@ -85,7 +85,9 @@ def train(self, shareable: Shareable, fl_ctx: FLContext, abort_signal: Signal) -
         validate_result: Shareable = self.learner.validate(shareable, fl_ctx, abort_signal)
 
         train_result = self.learner.train(shareable, fl_ctx, abort_signal)
-        if not (train_result and isinstance(train_result, Shareable)):
+        if not (
+            train_result and isinstance(train_result, Shareable) and train_result.get_return_code() == ReturnCode.OK
+        ):
```

Comment:
In the case that the train_result is already a valid Shareable but not OK, we should just return it. Why do we want to change the return code to EMPTY_RESULT?

Reply:
even though the returned is a valid Shareable, but "from_shareable(train_result)" may still fail, because the content may not be valid DXO. Maybe need further enhanced check here.

Reply:
The content doesn't have to contain a DXO, though we encourage people to do it. If the LearnerExecutor requires that (does it?), then it should check it explicitly, even if the return code is OK.

Reply:
Changed the check the "from_shareable(train_result)" explicitly.